test/core/xds/xds_metadata_test.cc

```diff
@@ -0,0 +1,276 @@
+//
+// Copyright 2022 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "src/core/xds/grpc/xds_metadata.h"
+
+#include <string>
+#include <utility>
+
+#include <google/protobuf/any.pb.h>
+#include <google/protobuf/struct.pb.h>
+
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+#include "absl/strings/str_format.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+#include "src/core/lib/debug/trace.h"
+#include "src/core/lib/gprpp/crash.h"
+#include "src/core/lib/gprpp/ref_counted_ptr.h"
+#include "src/core/util/json/json.h"
+#include "src/core/util/json/json_writer.h"
+#include "src/core/xds/grpc/xds_bootstrap_grpc.h"
+#include "src/core/xds/grpc/xds_metadata_parser.h"
+#include "src/core/xds/xds_client/xds_bootstrap.h"
+#include "src/core/xds/xds_client/xds_client.h"
+#include "src/core/xds/xds_client/xds_resource_type.h"
+#include "src/proto/grpc/testing/xds/v3/base.pb.h"
+#include "src/proto/grpc/testing/xds/v3/gcp_authn.pb.h"
+#include "test/core/test_util/scoped_env_var.h"
+#include "test/core/test_util/test_config.h"
+
+using envoy::config::core::v3::Metadata;
+using envoy::extensions::filters::http::gcp_authn::v3::Audience;
+
+namespace grpc_core {
+namespace testing {
+namespace {
+
+class XdsMetadataTest : public ::testing::Test {
+ protected:
+  XdsMetadataTest()
+      : xds_client_(MakeXdsClient()),
+        decode_context_{xds_client_.get(),
+                        *xds_client_->bootstrap().servers().front(),
+                        // FIXME: why does each test need its own tracer?
```

Comment:
was this meant to be fixed in this PR?

Reply:
Whoops!  Yes, I meant to fix this as part of this PR, but I somehow managed to do it on the wrong branch.  I've added the fix.